mama/c_cpp/src/c/clientmanageresponder.c

```diff
@@ -90,231 +90,234 @@ static void commandDestroyCB (wList list, void *element, void *closure)
-	{
-		/* The same callbacks will be sent to each subscriber regardless of the bridge
-		 * transport being used.
-		 */
-		mamaMsgCallbacks callbacks;
-		memset(&callbacks, 0, sizeof(callbacks));
-		callbacks.onCreate  = createCB;
-		callbacks.onError   = errorCB;
-		callbacks.onMsg     = msgCB;    
+    /* Get the default event queue from the bridgeImpl. */
+    mamaBridgeImpl *bridgeImpl = mamaTransportImpl_getBridgeImpl(impl->mTransport);
+    if(bridgeImpl != NULL)
+    {
+        /* The same callbacks will be sent to each subscriber regardless of the bridge
+         * transport being used.
+         */
+        mamaMsgCallbacks callbacks;
+        memset(&callbacks, 0, sizeof(callbacks));
+        callbacks.onCreate  = createCB;
+        callbacks.onError   = errorCB;
+        callbacks.onMsg     = msgCB;    
         callbacks.onDestroy = destroyCB;
 
         ret = mamaBridgeImpl_getInternalEventQueue((mamaBridge)bridgeImpl, &internalQueue);
-		if (ret == MAMA_STATUS_OK)
-		{
-			/* Enumerate all the sub transport bridges in the transport. */
-			int nextTransportIndex				= 0;
-			mamaPublisher *nextPublisher		= impl->publishers;
-			mamaSubscription *nextSubscription	= impl->subscriptions;	
-			while((ret == MAMA_STATUS_OK) && (nextTransportIndex < impl->numberTransportBridges))
-			{
-				/* The publisher allows commands to respond to point to point requests
-				 * without creating a new publisher, this must be created using the correct
-				 * transport bridge.
-				 */
-				ret = mamaPublisher_createByIndex(
-					nextPublisher,
-					impl->mTransport,
-					nextTransportIndex,
-					MAMA_CM_PUB_TOPIC,
-					NULL,
-					NULL);
-				if(ret == MAMA_STATUS_OK)
-				{
-					/* Allocate the subscription */
-					ret = mamaSubscription_allocate(nextSubscription);
-					if(ret == MAMA_STATUS_OK)
-					{
-						/* Set the subscription's transport index to ensure that the right transport is used. */
-						mamaSubscription_setTransportIndex(*nextSubscription, nextTransportIndex);
-			      
-						/* Create the subscription */
-						ret = mamaSubscription_createBasic(
-							*nextSubscription, 
-							impl->mTransport, 
-							internalQueue, 
-							&callbacks, 
-							MAMA_CM_TOPIC, 
-							impl);
+        if (ret == MAMA_STATUS_OK)
+        {
+            /* Enumerate all the sub transport bridges in the transport. */
+            int nextTransportIndex                = 0;
+            mamaPublisher *nextPublisher        = impl->publishers;
+            mamaSubscription *nextSubscription    = impl->subscriptions;    
+            while((ret == MAMA_STATUS_OK) && (nextTransportIndex < impl->numberTransportBridges))
+            {
+                /* The publisher allows commands to respond to point to point requests
+                 * without creating a new publisher, this must be created using the correct
+                 * transport bridge.
+                 */
+                ret = mamaPublisher_createByIndex(
```

Comment:
I was about to complain that this function is public and therefore you shouldn't change it but it appears that is it not, despite it's public-like name. While you're changing the signature to that function which will break any other implementation that depend on this function anyway, could you refactor it to be named mamaPublisherImpl_createByIndex as well please just to make it a little obvious that this is an internal function?